### ISSUE TITLE
Why does PrimeNg expect a router?  How can I workaround this in beta14?

### ISSUE THREAD

**Issue by JakeSummers** [primefaces/primeng#831]
Good Morning,

_Question 1:_ I am not using a router in my Application, but PrimeNg expects a router to be injected.  Why is this?  Could this be made an [optional provider](http://blog.thoughtram.io/angular/2015/05/18/dependency-injection-in-angular-2.html#optional-dependencies)?

_Question 2:_: What is the easiest way to pass in a dummy router? 

In PrimeNG 1.0.0-beta.12 and below, I could do the following, to trick the components into thinking that there was a router: 

```
@Component({
    selector: 'my-thing',
    template: require( './thing.component.html' ),
    directives: [Tree, ContextMenu ],
    providers: [{provide: Router, useValue: {}}]
})
export class Thing implements OnInit {
```

In PrimeNG 1.0.0-beta.13, I moved the Router into the AppModule and the trick continued to work:

```
@NgModule({
    bootstrap: [AppComponent],
    declarations: [
        ...
    ],
    imports: [
        ...
        RouterModule,
        DialogModule,
        ...
    ],
    providers: [ // expose our Services and Providers into Angular's dependency injection
        ...
        {provide: Router, useValue: {}}
        ...
    ]
})
export class AppModule {
}
```

When I upgrade to PrimeNG 1.0.0-beta.14 this trick no longer works.  What is the cleanest workaround?

**Comment by cagataycivici** [comment]
That is required for menu components to support routerLink functionality so that clicking on an item navigates to a view defined by the router, we are open to suggestions though. Not sure about dummy router.

**Comment by tanspac** [comment]
I ran into a similar issue. In my case my application is not using routing. I agree that router should be made optional since there is only one place in the code where the router is actually used. To work around this I created my own router service:

`export class Router
{
   constructor(){}
   public navigate(route: any[]):void
   {
       // Do nothing
   }
}`
Then I modified my `systemjs.config.js` to point `@angular/router` to my new router. This worked great and I can use the component without needing the router.
